src/App.js

```diff
@@ -17,19 +17,30 @@ export default class App extends Component {
   constructor(props) {
     super(props)
     this.state = {
+      authed: false,
       recommendationData: undefined,
       foodSearchTerm: '',
       showFilter: false,
       dietFilter: '',
       totalCount: 0,
     }
     this.toggleFilterMenu = this.toggleFilterMenu.bind(this)
+    this.onLoginClick = this.onLoginClick.bind(this)
+    this.onLogoutClick = this.onLogoutClick.bind(this)
     this.sendRecommendationRequest = this.sendRecommendationRequest.bind(this)
     this.setFoodAndMakeApiCall = this.setFoodAndMakeApiCall.bind(this)
     this.getRecoomendationListWithDietFilter = this.getRecoomendationListWithDietFilter.bind(this)
     this.sendRecommendationRequest = debounce(this.sendRecommendationRequest, 300)
   }
 
+  onLoginClick() {
```

Comment:
So doing onLoginClick = () = { // logic here } should give you access to the this scope of the component, thus not having to set .bind() within the constructor.

Reply:
☝️